compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -159,6 +159,8 @@ trait QuotesAndSplices {
       case _ => TypeBounds.empty
     val typeSym = newSymbol(spliceOwner(ctx), name, EmptyFlags, typeSymInfo, NoSymbol, tree.span)
     typeSym.addAnnotation(Annotation(New(ref(defn.QuotedRuntimePatterns_patternTypeAnnot.typeRef)).withSpan(tree.span)))
+    if !(typeSymInfo =:= TypeBounds.empty) then
+      report.warning(em"Type variable `$tree` has partially inferred bounds$pt.\n\nConsider defining bounds explicitly `'{ $typeSym$pt; ... }`", tree.srcPos)
```

Comment:
We should write `'[ type t; ... ]` for type patterns once we support https://github.com/lampepfl/dotty/pull/16942